content/guides/java/inject-extract.md

```diff
@@ -2,8 +2,142 @@
 title: "Java: Inject and Extract"
 ---
 
-* When do you use Inject and Extract?
-* Carriers and Format Options
-* Injecting and Extracting HTTP
-* Injecting and Extracting TextMap
-* Injecting and Extracting Binary
+## Introduction
+
+In order to trace across process boundaries and RPC calls in distributed systems, span context needs to propagated over the wire. The Java Opentracing API provides two functions in the Tracer interface to do just that, `inject(SpanContext, format, carrier)` and `extract(format, carrier)`.
+
+## Format Options and Carriers
+The **format** parameter refers to one of the three standard encodings the OpenTracing API defines:
+
+1. `TEXT_MAP` where span context is encoded as a collection of string key-value pairs,
+2. `BINARY` where span context is encoded as an opaque byte array,
+3. `HTTP_HEADERS`, which is similar to `TEXT_MAP` except that the keys must be safe to be used as HTTP headers.
+
+The **carrier** is an abstraction over the underlying RPC framework. For example, a carrier for `TEXT_MAP` format is an interface that allows the tracer to write key-value pairs via `put(key, value)` method, while a carrier for Binary format is simply a ByteBuffer.
+
+## Injecting and Extracting HTTP
+The following section discusses injecting and extracting a span's context over HTTP requests.
+#### Injecting the Span’s Context to HTTP
+In order to pass a span’s context over the HTTP request, the developer needs to call the `tracer.inject` before building the HTTP request, like so:
+
+```java
+Tags.SPAN_KIND.set(tracer.activeSpan(), Tags.SPAN_KIND_CLIENT);
+Tags.HTTP_METHOD.set(tracer.activeSpan(), "GET");
+Tags.HTTP_URL.set(tracer.activeSpan(), url.toString());
+tracer.inject(tracer.activeSpan().context(), Format.Builtin.HTTP_HEADERS, new RequestBuilderCarrier(requestBuilder));
+```
+Notice that a couple additional tags have been added to the `span` with some metadata about the HTTP request, following the OpenTracing Semantic Conventions.
```

Comment:
Link to https://github.com/opentracing/specification/blob/master/semantic_conventions.md